azurelinuxagent/common/protocol/wire.py

```diff
@@ -911,6 +917,43 @@ def get_header_for_cert(self):
             "x-ms-guest-agent-public-x509-cert": cert
         }
 
+    def get_host_plugin(self):
+        if self.host_plugin is None:
+            self.host_plugin = HostPluginProtocol(self.endpoint, self.get_goal_state())
+        return self.host_plugin
```

Comment:
Talked with Zhidong and apparently we should support GoalState change as RoleConfig name will change along with GoalState incarnation number.